test/abort/test-abort-backtrace.js

```diff
@@ -10,13 +8,19 @@ if (process.argv[2] === 'child') {
   process.abort();
 } else {
   const child = cp.spawnSync(`${process.execPath}`, [`${__filename}`, 'child']);
-  const frames =
-      child.stderr.toString().trimRight().split('\n').map((s) => s.trim());
+  const stderr = child.stderr.toString();
+  const frames = stderr.trimRight().split('\n').map((s) => s.trim());
 
   assert.strictEqual(child.stdout.toString(), '');
-  assert.ok(frames.length > 0);
-  // All frames should start with a frame number.
-  assert.ok(frames.every((frame, index) => frame.startsWith(`${index + 1}:`)));
-  // At least some of the frames should include the binary name.
-  assert.ok(frames.some((frame) => frame.includes(`[${process.execPath}]`)));
+  // `frames` will be empty for systems that don't support backtraces.
+  if (frames.length > 0) {
+
+    if (!frames.every((frame, index) => frame.startsWith(`${index + 1}:`))) {
+      assert.fail(`Each frame should start with a frame number:\n${stderr}`);
+    }
+
+    if (!frames.some((frame) => frame.includes(`[${process.execPath}]`))) {
+      assert.fail(`Some frames should include the binary name:\n${stderr}`);
+    }
+  }
```

Comment:
If the number of frames is lesser than or equal to zero, wouldn't it be better to log it as ignored?

Reply:
You mean call `common.skip()`? Or something else?

Reply:
IMHO we want to assert that that behavior is consistent.

P.S. Need to open an issue for documenting this behavior in:
https://github.com/nodejs/node/blob/99c478eb3629292e625a4d2fd634e89f0120ab91/doc/api/process.md#L406-L413

Reply:
> IMHO we want to assert that that behavior is consistent.

Assert that what behavior is consistent?

IIUC, there's no way to guarantee strictly from JS-land that a backtrace will be printed. It depends on which libc is used.

Reply:
@Trott Yes, I meant `common.skip` only, because the test is actually not successful but we are skipping.

Reply:
> @Trott Yes, I meant common.skip only, because the test is actually not successful but we are skipping. 

Perhaps. I was thinking that we're still testing that we aren't getting a malformed backtrace. A well-formed backtrace is fine. No backtrace is fine. Malformed backtrace is a problem. But I could go either way if you think the case is stronger for using `common.skip()` here.

Reply:
The comment above the `if` block says few systems don't support backtraces. We have a `common.skip` at the top of the file, for Windows. They both are the same?

Reply:
Windows and AIX don't support it ever, but for everything else, it depends on which libc is used. See https://github.com/nodejs/node/blob/f912080bf2d8fd024d3443f3ab9e399bc84a724b/src/backtrace_posix.cc#L7-L13

Reply:
Oh. Okay. Perhaps, we can set the skip message to Node.js is not compiled with GLIBC or something, like we do for OpenSSL.

Or lets not worry too much about it. The changes are good the way they are anyway. 👍